Best practices/Serialization & Deserialization.md

```diff
@@ -0,0 +1,106 @@
+
+To convert a JSON string back into an object:
+
+	YourObjectType obj = JsonSerializer.Deserialize<YourObjectType>(jsonString);
+	
+	
+### JsonSerializerOptions
+`JsonSerializerOptions` is a class that provides a way to specify various serialization and deserialization behaviors. Here are some of the commonly used options:
+
+**PropertyNameCaseInsensitive**: When set to `true`, property name matching during deserialization is case-insensitive. This can be useful when dealing with JSON from sources that don't maintain consistent casing.
+    
+    var options = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
+
+ **WriteIndented**: Determines whether the output JSON should be pretty-printed with indentation, which can be useful for debugging or human readability.
+
+    var options = new JsonSerializerOptions { WriteIndented = true }; 
+    
+**DefaultIgnoreCondition**: Determines the condition under which a property will be ignored during serialization. For instance, you can set it to `JsonIgnoreCondition.WhenWritingNull` to ignore properties with `null` values.
+   
+    var options = new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull }; 
+        
+**AllowTrailingCommas**: When set to `true`, the deserializer will tolerate trailing commas in the JSON.
+    
+    var options = new JsonSerializerOptions { AllowTrailingCommas = true }; 
+    
+**MaxDepth**: Sets the maximum depth allowed when reading or writing JSON, protecting against deeply nested structures.
+    
+    var options = new JsonSerializerOptions { MaxDepth = 10 }; 
+    
+**ReferenceHandler**: Determines how to handle references, which can be useful for dealing with circular references or preserving object references.
+    
+    var options = new JsonSerializerOptions { ReferenceHandler = ReferenceHandler.Preserve };
+    
+**Converters**: You can add custom converters to handle specific serialization or deserialization scenarios that aren't covered by the default behavior.
+    
+    options.Converters.Add(new MyCustomConverter()); 
+    
+These are just some of the many options available in `JsonSerializerOptions`. By understanding and utilizing these options, developers can fine-tune the behavior of `System.Text.Json` to match the specific needs of their applications.
+
+### Setting up JsonSerializerOptions globally
+
+The `AddJsonOptions` extension method allows you to configure serialization settings globally for all controllers in an ASP.NET Core application.  You typically do this in the `Program.cs` while configuring MVC services. Example:
+
+	services.AddControllers()
+		.AddJsonOptions(options => { ... });
+
+## Source Generators
+
+Starting from .NET 6, `System.Text.Json` introduced source generators, enhancing serialization performance. This feature can operate in two primary modes: gathering metadata and optimizing serialization. By default it utilises both modes.
```

Comment:
Should we add a couple of sentences describing the motivation for developing Source Generators for serialization?

We should also write about why aren't we using them by default, and in what cases are they useful.